pkg/utils/filesys/filesys.go

```diff
@@ -0,0 +1,44 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filesys
+
+import (
+	"context"
+	"io/fs"
+
+	"github.com/spf13/afero"
+)
+
+type contextKey string
+
+const (
+	KarpenterFS contextKey = "karpenter.sh/fs"
```

Comment:
consider making this a private variable. I also like to avoid referencing Karpenter anywhere in the project, since names can always change. Consider making this its own package inside utils like `filesystem`
```
filesystem.For(ctx)
filesystem.Inject(ctx)
```

Reply:
Ok on the first point.

> Consider making this its own package inside utils like filesystem [...]

I'm confused here - this already is a package under `utils` and usage reads `filesys.For(ctx)`?

Reply:
Oh wild -- I was misreading. You're totally correct XD. Any reason you chose filesys over filesystem for naming?

Reply:
No real reason, `filesystem` sounds better to me.

Reply:
Done